### PR TITLE
Redis pipelines + bonus

### PR DESCRIPTION
**Pull request by spladug**
:eyeglasses: @kjoconnor @dellis23

### CODE REVIEW

File: baseplate/context/redis.py [hunk 1]
```diff
@@ -4,55 +4,83 @@
...
-        raise NotImplementedError
+    def pipeline(self, name, transaction=True, shard_hint=None):
+        """Create a pipeline.
+
+        This returns an object on which you can call the standard redis
+        commands. Execution will be deferred until ``execute`` is called. This
+        is useful for saving round trips.
+
+        :param str name: The name to attach to diagnostics for this pipeline.
+        :param bool transaction: Whether or not the commands in the pipeline
```
Comment (kjoconnor) [inline baseplate/context/redis.py]: So will this work currently? I see `NotImplementedError`, but is that only if you're using `pipeline` directly?

**Comment by kjoconnor** [comment]
EXCELLENT

💅

File: baseplate/context/redis.py (hunk 1 above)
Comment (spladug) [inline baseplate/context/redis.py]: That other `transaction` method is separate. This will work.

**Comment by spladug** [comment]
:haircut:

**Comment by dellis23** [comment]
🐟  More diagnostics!  Yay!

File: baseplate/context/redis.py [hunk 2]
```diff
@@ -4,55 +4,83 @@
...
+    """Redis connection that collects diagnostic information.
 
+    This connection acts like :py:class:`redis.StrictRedis` except that all
+    operations are automatically wrapped with diagnostic collection.
+
+    The interface is the same as that class except for the
+    :py:meth:`~baseplate.context.redis.MonitoredRedisConnection.pipeline`
+    method.
+
+    .. note:: Locks and pubsub are currently unsupported.
```
Comment (kjoconnor) [inline baseplate/context/redis.py]: According to your other note transactions are still unsupported right? Only in pipelines?

**Comment by kjoconnor** [comment]
I am excite 🐟

**Pull request merged**